google/cloud/aiplatform/metadata/schema/google/artifact_schema.py

```diff
@@ -0,0 +1,248 @@
+            managed by clients (such as Vertex AI
+            Pipelines), and the system does not prescribe or
+            check the validity of state transitions.
+        """
+        extended_metadata = metadata or {}
+
+        super(VertexEndpoint, self).__init__(
+            schema_title=self.SCHEMA_TITLE,
+            resource_id=vertex_endpoint_id,
+            uri=uri,
+            display_name=display_name,
+            schema_version=schema_version,
+            description=description,
+            metadata=extended_metadata,
+            state=state,
+        )
+
+
+class UnmanagedContainerModel(base_artifact.BaseArtifactSchema):
+    """An artifact representing a Vertex Unmanaged Container Model."""
+
+    SCHEMA_TITLE = "google.UnmanagedContainerModel"
+
+    def __init__(
+        self,
+        predict_schema_ta: utils.PredictSchemata,
+        container_spec: utils.ContainerSpec,
+        unmanaged_container_model_id: Optional[str] = None,
+        uri: Optional[str] = None,
+        display_name: Optional[str] = None,
+        schema_version: Optional[str] = None,
+        description: Optional[str] = None,
+        metadata: Optional[Dict] = None,
+        state: gca_artifact.Artifact.State = gca_artifact.Artifact.State.LIVE,
+    ):
+        """Args:
+        predict_schema_ta (PredictSchemata):
+            An instance of PredictSchemata which holds instance, parameter and prediction schema uris.
+        container_spec (ContainerSpec):
+            An instance of ContainerSpec which holds the container configuration for the model.
+        unmanaged_container_model_id (str):
+            Optional. The <resource_id> portion of the Artifact name with
+            the format. This is globally unique in a metadataStore:
+            projects/123/locations/us-central1/metadataStores/<metadata_store_id>/artifacts/<resource_id>.
+        uri (str):
+            Optional. The uniform resource identifier of the artifact file. May be empty if there is no actual
+            artifact file.
+        display_name (str):
+            Optional. The user-defined name of the Artifact.
+        schema_version (str):
+            Optional. schema_version specifies the version used by the Artifact.
+            If not set, defaults to use the latest version.
+        description (str):
+            Optional. Describes the purpose of the Artifact to be created.
+        metadata (Dict):
+            Optional. Contains the metadata information that will be stored in the Artifact.
+        state (google.cloud.gapic.types.Artifact.State):
+            Optional. The state of this Artifact. This is a
+            property of the Artifact, and does not imply or
+            capture any ongoing process. This property is
+            managed by clients (such as Vertex AI
+            Pipelines), and the system does not prescribe or
+            check the validity of state transitions.
+        """
+        extended_metadata = metadata or {}
```

Comment:
I wonder if we should copy extended metadata throughout since we mutate it in place.

Reply:
Just to make  sure I'm following, you mean to use 
```python
extended_metadata = metadata.copy()
... 
```
instead of using by reference?

Reply:
done.